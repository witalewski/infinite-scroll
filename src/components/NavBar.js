import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { inject, observer } from 'mobx-react';
export const NavBar = ({ location, favourites }) => (
  <nav className="navbar sticky-top navbar-dark bg-dark navbar-expand-sm">
    <a className="navbar-brand" href="/">
      Infinite Scroll Demo
    </a>
    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarTogglerDemo01" aria-controls="navbarTogglerDemo01" aria-expanded="false" aria-label="Toggle navigation">
    <span class="navbar-toggler-icon"></span>
  </button>
    <div className="collapse navbar-collapse" id="navbarTogglerDemo01">
      <ul className="navbar-nav ">
        {[
          { label: 'New Photos', path: '/' },
          { label: `Favourites (${favourites.size})`, path: '/favourites' },
        ].map(item => (
          <li
            key={item.label}
            className={`nav-item ${location.pathname === item.path &&
              'active'}`}
          >
            <Link className="nav-link" to={item.path}>
              {item.label}
            </Link>
          </li>
        ))}
      </ul>
    </div>
  </nav>
);

export default withRouter(
  inject(({ appState }) => ({
    favourites: appState.favourites,
  }))(observer(NavBar))
);
