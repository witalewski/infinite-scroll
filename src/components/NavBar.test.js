import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockImageURLs from '../fixtures/imageURLs.json';
import { NavBar } from './NavBar';

describe('NavBar', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <NavBar
        favourites={new Set(mockImageURLs)}
        location={{ pathname: '/' }}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
