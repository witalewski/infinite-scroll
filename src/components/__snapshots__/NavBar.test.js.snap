// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<nav
  className="navbar sticky-top navbar-dark bg-dark navbar-expand-sm"
>
  <a
    className="navbar-brand"
    href="/"
  >
    Infinite Scroll Demo
  </a>
  <button
    aria-controls="navbarTogglerDemo01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarTogglerDemo01"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarTogglerDemo01"
  >
    <ul
      className="navbar-nav"
    >
      <li
        className="nav-item active"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/"
        >
          New Photos
        </Link>
      </li>
      <li
        className="nav-item false"
      >
        <Link
          className="nav-link"
          replace={false}
          to="/favourites"
        >
          Favourites (16)
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
